src/expressions/projectionExpressionChecker.js

```diff
@@ -1,39 +1,270 @@
 import { BaseSubExpressionChecker } from './baseSubExpressionChecker'
+import { wrapWithDefault } from '../defaultFactResolver'
+import { DISCIPL_FLINT_FACTS_SUPPLIED } from '../index'
 
 export class ProjectionExpressionChecker extends BaseSubExpressionChecker {
+  /**
+   * Create a SubExpressionChecker
+   * @param {ServiceProvider} serviceProvider
+   */
+  constructor (serviceProvider) {
+    super(serviceProvider)
+    this.scopeCheckers = {
+      'single': new SingleScopeProjectionExpressionChecker(serviceProvider),
+      'all': new AllScopeProjectionExpressionChecker(serviceProvider),
+      'some': new SomeScopeProjectionExpressionChecker(serviceProvider)
+    }
+  }
+
   async checkSubExpression (fact, ssid, context) {
-    const core = this._getFactChecker().getAbundanceService().getCoreAPI()
-    const lawregContext = context
+    const scope = fact.scope ? fact.scope : 'single'
+    this.logger.debug('Handling PROJECTION Expression with', scope, 'scope')
+    return this.scopeCheckers[scope].checkSubExpression(fact, ssid, context)
+  }
+}
 
+class BaseScopeProjectionExpressionChecker extends BaseSubExpressionChecker {
+  _checkProjectionIsValid (fact) {
     if (!fact.context || fact.context.length === 0) {
       throw new Error('A \'context\' array must be given for the PROJECTION expression')
     }
+    if (!fact.operand) {
+      // TODO deprecate projection expression fact property
+      if (fact.fact !== undefined) {
+        fact.operand = fact.fact
+      } else {
+        throw new Error('A \'operand\' must be given for the PROJECTION expression')
+      }
+    }
+  }
 
-    lawregContext.searchingFor = fact.fact
-    const initialLink = await this._getFactChecker().checkFact(fact.context[0], ssid, lawregContext)
-    const caseLink = await fact.context.slice(1).reduce(async (previousCaseLink, currentContextFact) => {
-      if (previousCaseLink) {
-        const factContext = await core.get(previousCaseLink, ssid)
+  /**
+   * Get filtered creating acts
+   * @param {object} fact - the create expression
+   * @param {ssid} ssid - Identifies the actor
+   * @param {Context} context - Context of the action
+   * @return {Promise<Object<string, Object<string, *>>>} - Object where key is the act link and value is the provided facts
```

Comment:
The we use this as an object anywhere? Maybe it makes sense for this to be an array of CreatingAct?